addons/payment_authorize/static/src/js/payment_form.js

```diff
@@ -139,39 +85,115 @@ const authorizeMixin = {
                 apiLoginID: this.authorizeInfo.login_id,
                 clientKey: this.authorizeInfo.client_key,
             },
-            ...this._getPaymentDetails(paymentOptionId),
+            ...this._getPaymentDetails(paymentOptionId, paymentMethodCode),
         };
 
         // Dispatch secure data to Authorize.Net to get a payment nonce in return
         Accept.dispatchData(
```

Comment:
It's not you who did this, but isn't it weird that we check the inputs (so far, so good), then trigger the payment, and THEN create the transaction? I believe that we should trigger the payment in an override of `_processDirectFlow` instead.